File: tests/rspec/pages/files/crop_file.rb
class CropFile < FileManagerPage
  set_url_matcher /files\/file\/crop/

  # Main box elements
  element :heading, 'div.form-standard form div.form-btns-top h1'
  element :crop_tab, 'ul.tabs a[rel="t-0"]'
  element :rotate_tab, 'ul.tabs a[rel="t-1"]'
  element :resize_tab, 'ul.tabs a[rel="t-2"]'
  element :save, 'div.form-standard form div.form-btns-top button'

  # Crop Form
  element :crop_width_input, 'form div.t-0 input[name="crop_width"]'
  element :crop_height_input, 'form div.t-0 input[name="crop_height"]'
  element :crop_x_input, 'form div.t-0 input[name="crop_x"]'
  element :crop_y_input, 'form div.t-0 input[name="crop_y"]'
  element :crop_image_preview, 'form div.t-0 figure.img-preview img'

  # Rotate Form
  element :rotate_right, 'form div.t-1 input[name="rotate"][value="270"]'
  element :rotate_left, 'form div.t-1 input[name="rotate"][value="90"]'
  element :flip_vertical, 'form div.t-1 input[name="rotate"][value="vrt"]'
  element :flip_horizontal, 'form div.t-1 input[name="rotate"][value="hor"]'
  element :rotate_image_preview, 'form div.t-1 figure.img-preview img'

  # Resize Form
  element :resize_width_input, 'form div.t-2 input[name="resize_width"]'
  element :resize_height_input, 'form div.t-2 input[name="resize_height"]'
  element :resize_image_preview, 'form div.t-2 figure.img-preview img'

  def load
    click_link 'Files'
    click_link 'About'
    file_name = find('div.box form table.app-listing tr:nth-child(2) td:first-child em').text
    find('div.box form table.app-listing tr:nth-child(2) td:nth-child(4) ul.toolbar li.crop').click

    return file_name
  end

end

File: tests/cypress/cypress/elements/pages/files/CropFile.js
import FileManagerSection from '../_sections/FileManagerSection'

class CropFile extends FileManagerSection {
  constructor() {
      super()
      this.urlMatch = /files\/file\/crop/;

      this.elements({
        // Main box elements
        'heading': '.ee-main .title-bar .title-bar__title',
        'crop_tab': '.tab-bar__tabs [rel="t-crop"]',
        'rotate_tab': '.tab-bar__tabs [rel="t-rotate"]',
        'resize_tab': '.tab-bar__tabs [rel="t-resize"]',
        'save': '.ee-main__content form .form-btns.form-btns-top button[value="save"]',

        // Crop Form
        'crop_width_input': 'form div.t-crop input[type!=hidden][name="crop_width"]',
        'crop_height_input': 'form div.t-crop input[type!=hidden][name="crop_height"]',
        'crop_x_input': 'form div.t-crop input[type!=hidden][name="crop_x"]',
        'crop_y_input': 'form div.t-crop input[type!=hidden][name="crop_y"]',
        'crop_image_preview': 'form .file-preview-modal__preview img',

        // Rotate Form
        'rotate_right': 'form div.t-rotate input[type!=hidden][name="rotate"][value="270"]',
        'rotate_left': 'form div.t-rotate input[type!=hidden][name="rotate"][value="90"]',
        'flip_vertical': 'form div.t-rotate input[type!=hidden][name="rotate"][value="vrt"]',
        'flip_horizontal': 'form div.t-rotate input[type!=hidden][name="rotate"][value="hor"]',
        'rotate_image_preview': 'form .file-preview-modal__preview img',

        // Resize Form
        'resize_width_input': 'form div.t-resize input[type!=hidden][name="resize_width"]',
        'resize_height_input': 'form div.t-resize input[type!=hidden][name="resize_height"]',
        'resize_image_preview': 'form .file-preview-modal__preview img',
      })
    }
    load() {
      cy.contains('Files').click()
      cy.get('.sidebar__link').contains('About').click()
      let filename = cy.get('.ee-main__content form .table-responsive table tr:nth-child(2) td:first-child em').invoke('text').then((text) => {
        console.log(text.trim())
        return text.trim()
      })
      cy.get('.ee-main__content form .table-responsive table tr:nth-child(2) td:nth-child(4) ul.toolbar li.crop').click()

      console.log(filename)

      return filename
    }
}
export default CropFile;